packages/@ourworldindata/grapher/src/entitySelector/EntitySelector.tsx

```diff
@@ -216,20 +235,121 @@ export class EntitySelector extends React.Component<{
+        return (
+            this.manager.entitySelectorState.isLoadingExternalSortColumn ??
+            false
+        )
+    }
+
     @computed private get table(): OwidTable {
         return this.manager.tableForSelection
     }
 
-    @computed private get sortColumns(): CoreColumn[] {
-        const activeSlugs = this.manager.activeColumnSlugs ?? []
+    @computed private get hasCountryOrRegionEntities(): boolean {
+        return (
+            intersection(
+                this.availableEntityNames,
+                regions.map((region) => region.name)
+            ).length > 0
+        )
+    }
 
-        const sortColumns = activeSlugs
+    @computed private get numericalChartColumns(): CoreColumn[] {
+        const activeSlugs = this.manager.activeColumnSlugs ?? []
+        return activeSlugs
             .map((slug) => this.table.get(slug))
             .filter(
                 (column) => column.hasNumberFormatting && !column.isProjection
             )
+    }
+
+    @computed
+    private get numericalChartColumnsWithoutPopulationAndGdpPerCapita(): CoreColumn[] {
+        return this.numericalChartColumns.filter(
+            (column) =>
+                column.slug !== this.populationSlug &&
+                column.slug !== this.gdpPerCapitaSlug
+        )
+    }
+
+    @computed private get sortColumns(): CoreColumn[] {
+        return excludeUndefined([
+            this.table.entityNameColumn,
+            this.populationColumn,
+            this.gdpPerCapitaColumn,
+            ...this.numericalChartColumnsWithoutPopulationAndGdpPerCapita,
+        ])
+    }
+
+    @computed private get populationColumnUsedInChart():
+        | CoreColumn
+        | undefined {
+        const activePopulationColumn = this.numericalChartColumns.find(
+            (column) =>
+                isPopulationVariableId(column.slug) ||
+                !!makeColumnLabel(column).match(/\bpopulation\b/i)
+        )
+        return activePopulationColumn
+    }
+
+    /**
+     * Either the ID of a population indicator used in the chart,
+     * or the default population indicator ID (or undefined if we
+     * don't want to sort by population)
+     */
+    @computed private get populationIndicatorId(): number | undefined {
+        if (this.populationColumnUsedInChart)
+            return +this.populationColumnUsedInChart.slug
+
+        if (this.hasCountryOrRegionEntities)
+            return DEFAULT_POPULATION_INDICATOR_ID
+
+        return undefined
+    }
+
+    @computed private get populationSlug(): ColumnSlug | undefined {
+        return this.populationIndicatorId !== undefined
+            ? this.populationIndicatorId.toString()
+            : undefined
+    }
+
+    @computed private get gdpPerCapitaColumnUsedInChart():
+        | CoreColumn
+        | undefined {
+        const activeGdpPerCapitaColumn = this.numericalChartColumns.find(
+            (column) => !!makeColumnLabel(column).match(/\bgdp per capita\b/i)
```

Comment:
Similar here: This'll also match any variable that contains `% of GDP per capita`, for example.

Reply:
GDP per capita is different because we have lots of GDP per capita variations like `GDP per capita (constant 2015 US$)` or `Real GDP per capita (2011$)` that are often simply referred to as "GDP per capita" in charts ([Datasette](https://datasette-public.owid.io/owid?sql=select%0D%0A++distinct+v.name,%0D%0A++v.id%0D%0Afrom%0D%0A++chart_dimensions+cd%0D%0A++join+variables+v+on+cd.variableId+%3D+v.id%0D%0Awhere%0D%0A++v.name+like+'%25gdp+per+capita%25'); [example chart](https://ourworldindata.org/grapher/agriculture-value-added-per-worker-vs-gdp-per-capita)). In such cases, using a second, slightly different GDP per capita version in the entity selector would be confusing. So, I think matching the name makes sense here. But I updated the regex to make sure that we're not matching "GDP per capita" if it's in parens, like `Government expenditure per student, primary (% of GDP per capita)`. 

There are some indicators that we'll match incorrectly (like `Growth rate of GDP per capita` or `Affordability of cigarettes: percentage of GDP per capita required to purchase 2000 cigarettes of the most sold brand`), which means that we'll fail to offer "GDP per capita" as a sorting option, which I think is acceptable.

Reply:
Yes, agreed. Good on you for pointing out that the outcome of a false positive is simply that we're not offering GDP per capita for sorting in that case, which is totally accepting.